### DISCUSSION TITLE
Webstorm crash if copilot enabled and two opened instances.

### DISCUSSION THREAD

**Discussion by dfiorentin** [community/community#128572] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi,
when using the jetbrains webstorm version 2024.1.4, with 2 different windows and projects opened, the whole application crash due to copilot.

Those are the version and logs.

copilot 1.5.8.5775
Webstorm 2024.1, Apple M3 macos 

This is the copilot log.
[threadDump-20240614-150226.txt](https://github.com/user-attachments/files/15838994/threadDump-20240614-150226.txt)

**Comment by ajwootto** [comment]
Seeing the same thing, each time I open a new project window it hangs the application in that window. It seems like the only way to get that project open is to close all other projects, invalidate caches and restart, and _then_ open it.

**Comment by plb666** [comment]
Same here. For a while now, the CoPilot plugin has been making the Webstorm unusable for some project: frequent freezes, abnormal CPU and memory consumption. Until today, I solved this by installing the May version of the plugin and everything worked properly again. With the new version of the IDE, 2024.2.1, this is no longer possible because it is no longer compatible. I had to deactivate the plugin in order to be able to work. Do you know of any work-arounds to get the system working again?

**Comment by plb666** [comment]
Here is an example of how much memory copilot requires in webstorm: 

<img width="873" alt="image" src="https://github.com/user-attachments/assets/8f8c1cff-5d7c-4a4e-890d-bce891f5754c">

Does it depend on how many files a project has?
